### PR TITLE
tests: Expand PVC resizing E2E to cover the walStorage

### LINKED ISSUE #676: Validate resizing of `walStorage` volume
Introduce E2E tests about resizing of the `walStorage` volume and use this as an opportunity to re-assess the documentation (maybe clarifying better this capability and how it works in the WAL volume section, if needed).

### PR DESCRIPTION
**Pull request by NiccoloFei**
Closes #676.

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

### CODE REVIEW

**Comment by armru** [comment]
do not squash the first two commits

**Review by armru — APPROVED** [review]

**Comment by armru** [comment]
another thing to keep in mind while squashing is that we should ensure that the commits have the PR reference (this probably would require a rebase with change of commits name)

**Review by armru — APPROVED** [review]

File: api/v1/cluster_webhook.go [hunk 1]
```diff
@@ -1129,15 +1129,42 @@ func (r *Cluster) validateStorageChange(old *Cluster) field.ErrorList {
...
+				r.Spec.WalStorage,
+				"walStorage cannot be disabled once the cluster is created"),
+		)
+	}
+
+	// we ensure that everything is equal except for the size
+	oldNormalized := old.DeepCopy()
+	oldNormalized.Spec.WalStorage.Size = ""
+	newNormalized := r.DeepCopy()
+	newNormalized.Spec.WalStorage.Size = ""
```
Comment (mnencia) [inline api/v1/cluster_webhook.go:1154]: The WalStoragea fields could be both nil here, granting a panic.

**Review by wadlejitendra — APPROVED** [review]

File: api/v1/cluster_webhook.go [hunk 2]
```diff
@@ -1127,17 +1127,48 @@ func (r *Cluster) validateStorageChange(old *Cluster) field.ErrorList {
...
+
 	var result field.ErrorList
 
-	if !reflect.DeepEqual(old.Spec.WalStorage, r.Spec.WalStorage) {
+	if old.Spec.WalStorage == nil && r.Spec.WalStorage != nil {
+		return append(result,
+			field.Invalid(
+				field.NewPath("spec", "walStorage"),
+				r.Spec.WalStorage,
+				"walStorage can be set only during the cluster creation"),
```
Comment (gbartolini) [inline api/v1/cluster_webhook.go:1141]:
```suggestion
				"walStorage can only be set at cluster creation"),
```

File: api/v1/cluster_webhook.go [hunk 3]
```diff
@@ -1127,17 +1127,48 @@ func (r *Cluster) validateStorageChange(old *Cluster) field.ErrorList {
...
+	if old.Spec.WalStorage != nil && r.Spec.WalStorage == nil {
+		return append(result,
+			field.Invalid(
+				field.NewPath("spec", "walStorage"),
+				r.Spec.WalStorage,
+				"walStorage cannot be disabled once the cluster is created"),
+		)
+	}
+
+	// we ensure that everything is equal except for the size
```
Comment (gbartolini) [inline api/v1/cluster_webhook.go:1154]:
```suggestion
	// We need to make sure that only the size can change
```

**Review by gbartolini — APPROVED** [review]

**Pull request merged**